From 258afbd776ef6fbe9e26947db1ed520685f728c8 Mon Sep 17 00:00:00 2001
From: Marcin Wojtas <mw@semihalf.com>
Date: Sun, 21 Jul 2019 10:21:31 +0200
Subject: [PATCH 353/386] telephony: mvebu_phone: use unique name for IRQ and
 procfs

In case of supporting the multiple controllers IRQ and
procfs entries should differ. Use unique device name
to achieve that.

Change-Id: I80dae956706a944dafda24a2297518909dafd2ff
Signed-off-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-on: https://sj1git1.cavium.com/12961
Tested-by: sa_ip-sw-jenkins <sa_ip-sw-jenkins@marvell.com>
Reviewed-by: Stefan Chulski <stefanc@marvell.com>
Reviewed-by: Yan Markman <ymarkman@marvell.com>
[RH: Original patch taken from marvell 88F3720 board support SDK 10.0-PR2003]
Signed-off-by: Ruiqiang Hao <Ruiqiang.Hao@windriver.com>
---
 drivers/telephony/mvebu_phone/mv_phone_dev.c | 7 ++++---
 1 file changed, 4 insertions(+), 3 deletions(-)

diff --git a/drivers/telephony/mvebu_phone/mv_phone_dev.c b/drivers/telephony/mvebu_phone/mv_phone_dev.c
index f7ee63c333d2..2f151e338428 100644
--- a/drivers/telephony/mvebu_phone/mv_phone_dev.c
+++ b/drivers/telephony/mvebu_phone/mv_phone_dev.c
@@ -562,7 +562,7 @@ int tdm_if_init(struct tal_params *tal_params)
 	}
 
 	/* Create TDM procfs statistics */
-	priv->tdm_stats = proc_mkdir("tdm", NULL);
+	priv->tdm_stats = proc_mkdir(dev_name(priv->dev), NULL);
 	if (priv->tdm_stats) {
 		if (!proc_create("tdm_stats", 0444,
 				 priv->tdm_stats, &mv_phone_operations))
@@ -572,7 +572,8 @@ int tdm_if_init(struct tal_params *tal_params)
 	/* Register TDM interrupts */
 	irqs_requested = 0;
 	for (i = 0; i < priv->irq_count; i++) {
-		ret = request_irq(priv->irq[i], tdm_if_isr, 0x0, "tdm", NULL);
+		ret = request_irq(priv->irq[i], tdm_if_isr, 0x0,
+				  dev_name(priv->dev), NULL);
 		if (ret) {
 			dev_err(priv->dev, "%s: Failed to connect irq(%d)\n",
 				__func__, priv->irq[i]);
@@ -651,7 +652,7 @@ void tdm_if_exit(void)
 
 		/* Remove proc directory & entries */
 		remove_proc_entry("tdm_stats", priv->tdm_stats);
-		remove_proc_entry("tdm", NULL);
+		remove_proc_entry(dev_name(priv->dev), NULL);
 
 		/* Release interrupt */
 		for (i = 0; i < priv->irq_count; i++)
-- 
2.17.1

